hr_expense_same_month/models/hr_expense_sheet.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+import logging
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class HrExpenseSheetConstrain(models.Model):
+    _inherit = 'hr.expense.sheet'
+
+    @api.constrains('expense_line_ids')
+    def _are_expenses_same_months(self):
+        for record in self:
+            months = {line.date.month for line in record.expense_line_ids}
```

Comment:
Hello @lanto-razafindrabe
we can use record.mapped('expense_line_ids.date.month'), it will return direction a liste of month. No need to add a loop.